tools/src/test/java/org/apache/kafka/tools/DelegationTokenCommandTest.java

```diff
@@ -0,0 +1,105 @@
+
+        // create token1 with renewer1
+        DelegationToken tokenCreated = DelegationTokenCommand.createToken(adminClient, getCreateOpts(renewer1));
+
+        List<DelegationToken> tokens = DelegationTokenCommand.describeToken(adminClient, getDescribeOpts(""));
+        assertEquals(1, tokens.size());
+        DelegationToken token1 = tokens.get(0);
+        assertEquals(token1, tokenCreated);
+
+        // create token2 with renewer2
+        DelegationToken token2 = DelegationTokenCommand.createToken(adminClient, getCreateOpts(renewer2));
+
+        tokens = DelegationTokenCommand.describeToken(adminClient, getDescribeOpts(""));
+        assertEquals(2, tokens.size());
+        assertEquals(Arrays.asList(token1, token2), tokens);
+
+        //get tokens for renewer2
+        tokens = DelegationTokenCommand.describeToken(adminClient, getDescribeOpts(renewer2));
+        assertEquals(1, tokens.size());
+        assertEquals(Collections.singletonList(token2), tokens);
+
+        //test renewing tokens
+        Long expiryTimestamp = DelegationTokenCommand.renewToken(adminClient, getRenewOpts(token1.hmacAsBase64String()));
+        DelegationToken renewedToken = DelegationTokenCommand.describeToken(adminClient, getDescribeOpts(renewer1)).get(0);
+        assertEquals(expiryTimestamp, renewedToken.tokenInfo().expiryTimestamp());
+
+        //test expire tokens
+        DelegationTokenCommand.expireToken(adminClient, getExpireOpts(token1.hmacAsBase64String()));
+        DelegationTokenCommand.expireToken(adminClient, getExpireOpts(token2.hmacAsBase64String()));
+
+        tokens = DelegationTokenCommand.describeToken(adminClient, getDescribeOpts(""));
+        assertEquals(0, tokens.size());
+
+        //create token with invalid renewer principal type
+        assertThrows(ExecutionException.class, () -> DelegationTokenCommand.createToken(adminClient, getCreateOpts("Group:Renewer3")));
+
+        // try describing tokens for unknown owner
+        assertTrue(DelegationTokenCommand.describeToken(adminClient, getDescribeOpts("User:Unknown")).isEmpty());
+
+    }
+
+    private DelegationTokenCommand.DelegationTokenCommandOptions getCreateOpts(String renewer) {
+        String[] args = {"--bootstrap-server", "localhost:9092", "--max-life-time-period", "-1", "--command-config", "testfile", "--create", "--renewer-principal", renewer};
+        return new DelegationTokenCommand.DelegationTokenCommandOptions(args);
+    }
+
+    private DelegationTokenCommand.DelegationTokenCommandOptions getDescribeOpts(String owner) {
+        String[] args;
+        if (!owner.equals("")) {
+            args = new String[] {"--bootstrap-server", "localhost:9092", "--command-config", "testfile", "--describe","--owner-principal", owner};
+        } else {
+            args = new String[] {"--bootstrap-server", "localhost:9092", "--command-config", "testfile", "--describe"};
+        }
+        return new DelegationTokenCommand.DelegationTokenCommandOptions(args);
```

Comment:
Maybe the following will save us a few repetitive strings?
```suggestion
        List<String> args = new ArrayList() {{
          .add("--bootstrap-server");
          .add("localhost:9092");
          .add("--command-config");
          .add("testfile");
          .add("--describe")
        }}
        if (!owner.equals("")) {
          args.add("--owner-principal");
          args.add(owner);
        }
        return new DelegationTokenCommand.DelegationTokenCommandOptions(args);
```